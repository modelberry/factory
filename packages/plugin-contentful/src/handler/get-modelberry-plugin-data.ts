import { ContentTypeFieldValidation } from 'contentful-management/types'
import { ModelberryVariable } from '@modelberry/mbfactory'

export type Validation = ContentTypeFieldValidation & {
  message?: string
  nodes?: any
}

export type ValidationsMap = {
  [validationName: string]: Validation
}

export type ModelberryPluginData = {
  '@modelberry/plugin-contentful/plain'?: {
    validations?: ValidationsMap
  }
}

export interface GetModelberryPluginData {
  dataVar: ModelberryVariable
}

export const getModelberryPluginData = ({
  dataVar,
}: GetModelberryPluginData) => {
  const dataVarFn = new Function(`return ${dataVar.value}`)
  const dataVarObj = dataVarFn() as ModelberryPluginData
  const data = dataVarObj?.['@modelberry/plugin-contentful/plain']

  return data
}
