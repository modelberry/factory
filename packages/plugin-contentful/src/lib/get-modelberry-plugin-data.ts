import { ContentTypeFieldValidation } from 'contentful-management/types'
import { ModelberryVariable } from '@modelberry/mbfactory/plain'

export type ValidationsMap = {
  [validationName: string]: ContentTypeFieldValidation
}

export type ModelberryPluginData = {
  '@modelberry/plugin-contentful/plain'?: {
    validations?: ValidationsMap
  }
}

export interface GetModelberryPluginData {
  dataVar: ModelberryVariable
}

export const getModelberryPluginData = ({
  dataVar,
}: GetModelberryPluginData) => {
  const dataVarFn = new Function(`return ${dataVar.value}`)
  const dataVarObj = dataVarFn() as ModelberryPluginData
  const data = dataVarObj?.['@modelberry/plugin-contentful/plain']

  return data
}
