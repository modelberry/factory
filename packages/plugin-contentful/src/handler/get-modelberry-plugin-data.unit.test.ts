import { getModelberryPluginData } from './get-modelberry-plugin-data'

const dataVar = {
  fileName: '/__fixtures__/topic.ts',
  isArray: false,
  isExported: true,
  isObject: true,
  isTyped: true,
  name: 'modelberryPluginData',
  type: 'ModelberryPluginData',
  value: `{
    '@modelberry/plugin-contentful': {
      validations: {
        dropdown: { in: ['item A', 'item B', 'item C'] },
        camelCase: {
          regexp: { pattern: '^[a-z]+([A-Z][a-z0-9]+)*$', flags: '' },
        },
      },
    },
  }`,
}

describe('getModelberryPluginData should', () => {
  afterEach(() => {
    jest.clearAllMocks()
  })

  test('get data correctly', async () => {
    const data = getModelberryPluginData({ dataVar })
    expect(data).toMatchSnapshot()
  })
})
