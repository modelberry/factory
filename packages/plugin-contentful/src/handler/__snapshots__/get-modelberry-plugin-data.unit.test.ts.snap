// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getModelberryPluginData should get data correctly 1`] = `
{
  "validations": {
    "camelCase": {
      "regexp": {
        "flags": "",
        "pattern": "^[a-z]+([A-Z][a-z0-9]+)*$",
      },
    },
    "dropdown": {
      "in": [
        "item A",
        "item B",
        "item C",
      ],
    },
  },
}
`;
